presto-main/src/main/java/com/facebook/presto/execution/UseTask.java

```diff
@@ -72,4 +77,16 @@ private void checkAndSetCatalog(Use statement, Metadata metadata, QueryStateMach
             stateMachine.setSetCatalog(catalog);
         }
     }
+
+    private void checkAndSetSchema(Use statement, Metadata metadata, QueryStateMachine stateMachine, Session session)
+    {
+        String catalog = statement.getCatalog()
+                .map(Identifier::getValueLowerCase)
+                .orElseGet(() -> session.getCatalog().map(String::toLowerCase).get());
+        String schema = statement.getSchema().getValueLowerCase();
+        if (!metadata.getMetadataResolver(session).schemaExists(new CatalogSchemaName(catalog, schema))) {
+            throw new PrestoException(NOT_FOUND, format("Schema does not exist: %s.%s", catalog, schema));
```

Comment:
Should this be SemanticException here? 
https://github.com/prestodb/presto/blob/master/presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticErrorCode.java#L29

Reply:
Sure, fixed. And fix the exception thrown by `checkAndSetCatalog` as well, please take a look when convenient, thanks.